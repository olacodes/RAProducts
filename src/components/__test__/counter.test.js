import React from "react";
import Counter from "../counter";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { NativeBaseProvider } from "native-base";

describe("Test Counter Component", () => {
  const initialState = { counter: { value: 10 } };
  const mockStore = configureStore();
  let store, wrapper;

  it(`renders correctly`, () => {
    store = mockStore(initialState);

    const tree = TestRenderer.create(
      <Provider store={store}>
        <NativeBaseProvider>
          <Counter />
        </NativeBaseProvider>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
