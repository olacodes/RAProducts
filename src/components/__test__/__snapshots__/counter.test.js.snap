// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Counter Component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
